modules/internal/ChapelDistribution.chpl

```diff
@@ -785,8 +803,11 @@ module ChapelDistribution {
     delete dist;
   }
 
-  proc _delete_dom(dom:BaseDom, param privatized:bool) {
-    dom.dsiDestroyDom();
+  proc _delete_dom(dom, param privatized:bool) {
+    use Reflection;
+    if canResolveMethod(dom, "dsiDestroyDom") {
+      dom.dsiDestroyDom();
```

Comment:
This conditional seems odd to me. Wouldn't it be an error not to implement dsiDestroyDom?